pallets/oracle/src/lib.rs

```diff
@@ -367,3 +367,19 @@ impl<T: Config> Pallet<T> {
 		T::DataProvider::get_no_op(key)
 	}
 }
+
+pub trait OracleApi<Balance, CurrencyId> {
+	fn currency_to_usd(
+		amount: &Balance,
+		currency_id: &CurrencyId,
+	) -> Result<Balance, DispatchError>;
+}
+
+impl<T: Config> OracleApi<BalanceOf<T>, CurrencyId> for Pallet<T> {
+	fn currency_to_usd(
+		amount: &BalanceOf<T>,
+		currency_id: &CurrencyId,
+	) -> Result<BalanceOf<T>, DispatchError> {
+		Pallet::<T>::currency_to_usd(amount.clone(), currency_id.clone())
+	}
+}
```

Comment:
Should we put this definition more to the top of the file or maybe even an extra file? It's a bit hidden down here